Tests/DatadogTests/Datadog/Mocks/CoreMocks.swift

```diff
@@ -1246,4 +1246,70 @@ class PrintFunctionMock {
     func print(message: String) {
         printedMessages.append(message)
     }
+
+    func reset() {
+        printedMessages = []
+    }
+}
+
+class CoreLoggerMock: CoreLoggerType {
```

Comment:
Yeah, I was thinking of just mocking the "print" function, instead of the entire logger, but I decided to go with `CoreLoggerMock` for 3 reasons:
* If we mock the entire "print", we would have to build assertions around formatted `String`, instead of `message:String` and `error:DDError`. That would require using regexps to shell out the interesting part in a `String` (and to ignore time format), so in overall the tests code will be more complex. In consequence, that might invite less strict assertions and adding weaker tests because of the additional overhead.
* If we mock the "print" function, then we don't need DI interface for the `CoreLogger`. We know that `CoreLogger` will appear in V2's public interface, so exposing the entire type in public might be less pragmatic than separating only its interface. 
* Last, `CoreLogger` + interface is aligned with `RUMTelemetry` + `Telemetry` (interface). Because we consider both to be bundled in `DD`, I think it makes sense to have this convention.

How does it sound @maxep ?